controlplane/internal/controller/clusterdeployment_controller.go

```diff
@@ -251,6 +266,20 @@ func computeAgentClusterInstall(clusterDeployment *hivev1.ClusterDeployment, acp
 	if cluster.Spec.ClusterNetwork != nil && cluster.Spec.ClusterNetwork.Services != nil {
 		serviceNetwork = cluster.Spec.ClusterNetwork.Services.CIDRBlocks
 	}
+	var additionalManifests []hiveext.ManifestsConfigMapReference
+	if len(acp.Spec.AgentConfigSpec.ManifestsConfigMapRefs) > 0 {
+		additionalManifests = append(additionalManifests, acp.Spec.AgentConfigSpec.ManifestsConfigMapRefs...)
+	}
+
+	if acp.Spec.AgentConfigSpec.ImageRegistryRef != nil {
```

Comment:
Can we extract image registry functionality to a different package, maybe `controlplane/internal/registry` or something similar? So that the body below can look something like:
```suggestion
	if acp.Spec.AgentConfigSpec.ImageRegistryRef != nil {
	
	     imageRegistryManifestRef, err := imageRegistry.CreateConfig(ctx, r.Client, acp.Spec.AgentConfigSpec.ImageRegistryRef)
	     
	     if err != nil {
	         // handle error
	     }
	     additionalManifests = append(additionalManifests, imageRegistryManifestRef)
	}
```

Reply:
Good idea, thank you! Will do!